### DISCUSSION TITLE
After docker save can't connect mongo!!

### DISCUSSION THREAD

**Discussion by ch-boogeyman** [docker-library/mongo#556] [General]
```shell
# save first
docker save -o ./mongo-test.tar mongo

# others load
docker load -i  ./mongo-test.tar
dcoker run -it -p 27017:27017 --name mongo-test mongo bash
```
## and then can't connect,see images

![0](https://user-images.githubusercontent.com/46711557/181669019-45356623-e53c-4432-ab2e-444a7d49db85.png)

**Comment by ch-boogeyman** [comment]
## WHY?
Either docker commit can't save the changes in mongo container!!
```shell
### raw
docker run -itd --name mongo -p 27017:27017 mongo --auth
db.createUser({ user:'admin',pwd:'123456',roles:[ { role:'userAdminAnyDatabase', db: 'admin'},"readWriteAnyDatabase"]});
docker commit 1a40a0e27f42 mongo-commit:v1

### new
docker run it  mongo-commit bash
# then no user in new container,I have set admin in old container...
```
![0](https://user-images.githubusercontent.com/46711557/181713931-ce37291a-7e36-4219-a6ec-d09790105683.png)
![1](https://user-images.githubusercontent.com/46711557/181713936-a9eca09b-11a2-4250-87e1-0ece0d27a12d.png)

**Comment by YowFung** [comment]
Hi, @ch-boogeyman 
Try using this command to create your docker container.
```shell
sudo docker run -d -p 27017:27017 --name mongo-test mongo mongod
```
Execute `sudo docker exec -it mongo-test bash` to enter this container, and then run `mongo` command in container.

**Comment by YowFung** [comment]
When a container is mirrored, the data it mounts on the host will not be attached to the mirror.

```shell
❯ docker inspect mongo-test | grep Mounts -A 21
        "Mounts": [
            {
                "Type": "volume",
                "Name": "3c9c2babfe7b189f401ea455c3ed11a0c2804737856de1c501d53bcbe31d99b7",
                "Source": "/var/lib/docker/volumes/3c9c2babfe7b189f401ea455c3ed11a0c2804737856de1c501d53bcbe31d99b7/_data",
                "Destination": "/data/configdb",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "volume",
                "Name": "0249ea8f46ab29e420d10f0ff1cf8ca344949e773d71cef7816460a62fa43d53",
                "Source": "/var/lib/docker/volumes/0249ea8f46ab29e420d10f0ff1cf8ca344949e773d71cef7816460a62fa43d53/_data",
                "Destination": "/data/db",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            }
        ],
```

Notice that the data you created or modified at your old container is saved in your host such as `/var/lib/docker/volumes/0249ea8f46ab29e420d10f0ff1cf8ca344949e773d71cef7816460a62fa43d53/_data`.

The mount directory of your new container is impossible keeping data from your old container.

**Comment by YowFung** [comment]
From a security point of view, you should not keep the database in the generated mirror.

If you want to use the same database in the front and back containers, you can specify the same volume path for both containers. For example:
```shell
sudo docker run -d -p 27017:27017 -v /data/mongo/db:/data/db -v /data/mongo/configdb:/data/configdb --name mongo-test mongo mongod
```

It is worth noting that, the `docker save` and `docker load` commands operate on mirrors instead of containers. If you want to mirror the containers, use the `docker commit` command.

**Comment by yosifkit** [comment]
Neither `docker commit` nor `docker save` will save the `VOLUME` contents, which is where the database files are stored. The simplest solution is to stop the container and then copy/zip/tar the database files to a new folder on your host (or a new named docker volume) and start the new one with that directory as the volume mount.

**Comment by ch-boogeyman** [comment]
Use docker `run -v`  to mount and then `docker commit` won't occurs the above problem~
